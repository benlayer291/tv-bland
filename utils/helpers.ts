export const removeProtocolFromUrl = (url: string): string =>
  url.replace(/^https?:\/\//i, '//')

export const removeHtmlTagsFromString = (htmlString: string): string =>
  htmlString.replace(/(<([^>]+)>)/gi, '')

export const checkDeviceIsMobile = (): boolean => {
  return (
    typeof navigator !== 'undefined' &&
    /Android|Mobi/i.test(navigator.userAgent)
  )
}
