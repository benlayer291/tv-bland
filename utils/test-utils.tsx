import { render, RenderOptions } from '@testing-library/react'

type Props = {
  children: React.ReactElement
}

const Providers: React.FunctionComponent<Props> = ({ children }: Props) => {
  return <>{children}</>
}

const customRender = (
  ui: React.ReactElement,
  options: Omit<RenderOptions, 'queries'> = {}
) => render(ui, { wrapper: Providers, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
