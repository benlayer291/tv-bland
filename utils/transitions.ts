type Transtion = {
  duration: number
  ease: number[]
}

export const transition: Transtion = {
  duration: 5,
  ease: [0.43, 0.13, 0.23, 0.96],
}
