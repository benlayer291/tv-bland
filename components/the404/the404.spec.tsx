import React from 'react'
import { render, RenderResult } from '../../utils/test-utils'

import The404 from './the404'

describe('<The404 />', () => {
  let el: RenderResult

  it('should render', () => {
    el = render(<The404 />)

    expect(el.container).toBeInTheDocument()
  })

  it('should match snapshot', () => {
    el = render(<The404 />)

    expect(el.container).toMatchSnapshot()
  })
})
