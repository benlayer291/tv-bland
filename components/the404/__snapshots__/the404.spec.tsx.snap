// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<The404 /> should match snapshot 1`] = `
<div>
  <div
    class="The404"
  >
    <div
      class="wrapper"
    >
      <div
        class="The404__text"
      >
        <h1>
          404
        </h1>
        <h2>
          Uh oh... looks like that wasn't scheduled ðŸ˜¬
        </h2>
        <p>
          <a
            href="/"
          >
            Return home
          </a>
        </p>
      </div>
    </div>
  </div>
</div>
`;
